spec.html

```diff
@@ -63,73 +63,125 @@ <h1>ApplyUnicodeExtensionToTag( _tag_, _options_, _relevantExtensionKeys_ )</h1>
+          1. If _e_ = -1, let _len_ be _size_ - _k_; else let _len_ be _e_ - _k_.
+          1. Let _subtag_ the String value equal to the substring of _extension_ consisting of the code units at indices _k_ (inclusive) through _k_ + _len_ (exclusive).
+          1. If _isKeyword_ is *false*, then
+            1. If _len_ &ne; 2 and _subtag_ is not an element of _attributes_, then
+              1. Append _subtag_ to _attributes_.
+          1. Else,
+            1. If _len_ = 2, then
+              1. If _keywords_ does not contain an element whose [[Key]] is the same as _key_, then
+                1. Append the Record{[[Key]]: _key_, [[Value]]: _value_} to _keywords_.
+            1. Else,
+              1. If _value_ is not the empty String, then
+                1. Let _value_ be the string-concatenation of _value_ and `"-"`.
+              1. Let _value_ be the string-concatenation of _value_ and _subtag_.
+          1. If _len_ = 2, then
+            1. Let _isKeyword_ be *true*.
+            1. Let _key_ be _subtag_.
+            1. Let _value_ be the empty String.
+          1. Let _k_ be _k_ + _len_ + 1.
+        1. If _isKeyword_ is *true*, then
+          1. If _keywords_ does not contain an element whose [[Key]] is the same as _key_, then
+            1. Append the Record{[[Key]]: _key_, [[Value]]: _value_} to _keywords_.
+        1. Return the Record{[[Attributes]]: _attributes_, [[Keywords]]: _keywords_}.
+      </emu-alg>
+    </emu-clause>
+
+    <emu-clause id="sec-canonicalize-unicode-extension" aoid=CanonicalizeUnicodeExtension>
+      <h1>CanonicalizeUnicodeExtension( _attributes_, _keywords_ )</h1>
+      <p>
+        The CanonicalizeUnicodeExtension abstract operation creates the canonical Unicode locale extension sequence from _attributes_, which must be a List of String values, and _keywords_, which must be a List of Record values. The empty String is returned if both arguments are empty Lists. The following steps are taken:
+      </p>
+
+      <emu-alg>
+        1. If _attributes_ is empty and _keywords_ is empty, then
+          1. Return the empty String.
+        1. Let _sortedAttributes_ be a new List containing the same values as the list _attributes_ where the values are ordered as if an Array of the same values had been sorted using `Array.prototype.sort` using *undefined* as _comparefn_.
+        1. Let _fullKeywords_ be the empty List.
+        1. Repeat for each element _entry_ of _keywords_ in List order,
+          1. Let _keyword_ be _entry_.[[Key]].
+          1. If _entry_.[[Value]] is not the empty String, then
+            1. Let _keyword_ be the string-concatenation of _keyword_, `"-"`, and _entry_.[[Value]].
+          1. Append _keyword_ to _fullKeywords_.
+        1. Let _sortedKeywords_ be a new List containing the same values as the list _fullKeywords_ where the values are ordered as if an Array of the same values had been sorted using `Array.prototype.sort` using *undefined* as _comparefn_.
```

Comment:
Nit: Technically, maybe this should be using a %-style reference to Array.prototype.sort. However, I'd be fine with more high-level wording such as "Let _sortedKeywords_ be a new List containing the same values as the list _fullKeywords_ where the values are ordered by ascending Unicode code point values, with shorter strings preceding longer ones."

Reply:
I was copying the style from [9.4.6.11 ModuleNamespaceCreate](https://tc39.github.io/ecma262/#sec-modulenamespacecreate), which also uses `Array.prototype.sort` instead of `%ArrayProto_sort%`. :smile: 

If it is acceptable for everyone, I'd like to replace step 2 with:

"Let _sortedAttributes_ be a new List containing the same values as the list _attributes_ where the values are ordered by ascending Unicode code point values, with shorter strings preceding longer ones."

And steps 3-5 with:

"Let _sortedKeywords_ be a new List containing the same values as the list _keyword_ where the values are ordered by ascending Unicode code point values of the [[Key]] field (or fields?), with shorter strings preceding longer ones."

Reply:
That wording looks great. Go for it. (No opinion on field vs fields, either seems fine.)